src/components/Filters.tsx

```diff
@@ -0,0 +1,85 @@
+import classNames from 'classnames';
+import React, { useState, useEffect, useContext } from 'react';
+import { NavLink } from 'react-router-dom';
+import { Context } from './Context';
+import { Todo } from '../types/Todo';
+
+export const Filters: React.FC = () => {
+  const { todos, setTodos } = useContext(Context);
+  const [shouldClearCompleted, setClearCompleted] = useState(true);
+  let activeTodos: Todo[] = [];
+  let completedTodos: Todo[] = [];
+
+  if (todos.length) {
+    activeTodos = todos.filter((todo: Todo) => todo.completed !== true);
+    completedTodos = todos.filter((todo: Todo) => todo.completed === true);
+  }
+
+  useEffect(() => {
+    if (activeTodos) {
+      setTodos(activeTodos);
+    }
+
+    if (activeTodos) {
+      setTodos(activeTodos);
+    }
+
+    setClearCompleted(true);
+  }, [shouldClearCompleted]);
+
+  const selectedClass = (isActive: boolean) => {
+    return (classNames({
+      selected: isActive,
+    }));
+  };
+
+  return (
+    <footer className="footer">
+      {typeof activeTodos !== 'undefined' && (
```

Comment:
```suggestion
      {activeTodos && (
```
also, if required, verify that incoming array is not empty